src/Components/Publishing/ToolTip/TooltipsDataLoader.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from "react"
+import PropTypes from "prop-types"
+import { QueryRenderer, graphql } from "react-relay"
+import { createEnvironment } from "../../../Relay/createEnvironment"
+import { ArticleData } from "../Typings"
+import { getArtsySlugsFromArticle } from "../Constants"
+import { keyBy } from "lodash"
+import { TooltipsDataLoaderQueryResponse } from "../../../__generated__/TooltipsDataLoaderQuery.graphql"
+
+interface Props {
+  article: ArticleData
+}
+
+// TODO: get enviroment from context or pass it in as prop
+const environment = createEnvironment()
+
+export class TooltipsDataLoader extends Component<Props> {
+  render() {
+    const { artists: artistSlugs } = getArtsySlugsFromArticle(
+      this.props.article
+    )
+
+    return (
+      <QueryRenderer
+        environment={environment}
+        query={graphql`
+          query TooltipsDataLoaderQuery($artistSlugs: [String!]) {
+            artists(slugs: $artistSlugs) {
+              id
+              ...ArtistToolTip_artist
+            }
+          }
+        `}
+        variables={{
+          artistSlugs,
+        }}
+        render={readyState => {
+          const data: TooltipsDataLoaderQueryResponse = { artists: [] }
+          Object.keys(readyState.props || {}).forEach(key => {
```

Comment:
Are you handling loading and error states properly by defaulting to an empty object?

Reply:
I'm not handling `readyState.error` explicitly, I was just relying on the fact that props is `null` if there's an error or the data is still loading and in that case we just don't render the tooltip.

Reply:
Gotcha 👍